src/mon/OSDMonitor.cc

```diff
@@ -7528,16 +7528,11 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
 
     CrushWrapper newcrush;
     _get_pending_crush(newcrush);
-
-    if (!newcrush.class_exists(srcname)) {
-      err = -ENOENT;
-      ss << "class '" << srcname << "' does not exist";
-      goto reply;
-    }
-
-    if (newcrush.class_exists(dstname)) {
-      err = -EEXIST;
-      ss << "class '" << dstname << "' already exists";
```

Comment:
the new condition logical doesn't match the original one, should it be changed from "&&" to "||" ?

Reply:
The CrushWrapper rename method catches the old errors that were caught here.  the new condition is making things idempotent (so a retry will succeed).